src/components/ConfigProvider/FeatureFlagProvider.tsx

```diff
@@ -0,0 +1,49 @@
+'use client';
+
+import React, { createContext, FC, useContext, useMemo } from 'react';
+
+export interface FeatureFlags {
+  panelLazyLoadContent?: boolean;
+}
+
+export interface FeatureFlagContextProps {
+  featureFlags?: FeatureFlags;
+  children?: React.ReactNode;
+}
+
+const FeatureFlagContext = createContext<FeatureFlags>(undefined);
+
+const defaultFeatureFlagValues: FeatureFlags = {
+  panelLazyLoadContent: false,
+};
+
+export const FeatureFlagContextProvider: FC<FeatureFlagContextProps> = ({
+  children,
+  featureFlags,
+}) => {
+  const ancestorFeatureFlags = useContext(FeatureFlagContext);
+
+  const currentContext = useMemo(
+    () => ({
+      ...(defaultFeatureFlagValues || {}),
+      ...(ancestorFeatureFlags || {}),
+      ...(featureFlags || {}),
+    }),
+    [featureFlags, ancestorFeatureFlags]
+  );
```

Comment:
Do we expect/support `featureFlags` changing between renders?
If this is just an application-level definition and not something loaded async from something like a backend config, we might be able to get a small optimization by wrapping it with a ref.

Also, not sure if this is a ES version-specific feature, but you can use the spread syntax on `undefined`
e.g.
```suggestion
  featureFlags,
}) => {
  const featureFlagsRef = useRef(featureFlags);
  const ancestorFeatureFlags = useContext(FeatureFlagContext);

  const currentContext = useMemo(
    () => ({
      ...defaultFeatureFlagValues,
      ...ancestorFeatureFlags,
      ...featureFlagsRef.current,
    }),
    [ancestorFeatureFlags]
  );
```

Reply:
I had a fallback below where I did this: 
```
  useContext(FeatureFlagContext) || { ...defaultFeatureFlagValues };
```
This gave me a fallback with the default values if I run the code without the config provider being set for some reason. Maybe there's a better way to manage this though?

Reply:
I would expect it to possibly be loaded async from a config, and possibly managed via gates, though I wouldn't expect it to change frequently at runtime.